cpp/include/cudf/copying.hpp

```diff
@@ -557,5 +557,68 @@ std::unique_ptr<column> copy_if_else(column_view const& lhs, scalar const& rhs,
 std::unique_ptr<column> copy_if_else( scalar const& lhs, scalar const& rhs, column_view const& boolean_mask,
                                     rmm::mr::device_memory_resource *mr = rmm::mr::get_default_resource());
 
+/**
+ * @brief Scatters dense `i`th buffer/column from `source` of size N to `i`th` target` column of
+ * size K (K >= N) as per `boolean_mask` of size K.
+ *
+ * `boolean_mask` should have N number of `true`s. If boolean mask is `true`,
+ * corresponding value in target is updated with value from corresponding
+ * `source` column, else it is left untouched.
+ *
+ * Example:
+ * source: {{1, 5, 6, 8, 9}}
+ * boolean_mask: {true, false, false, false, true, true, false, true, true, false}
+ * target:       {{   2,     2,     3,     4,    4,     7,    7,    7,    8,    10}}
+ *
+ * output:       {{   1,     2,     3,     4,    5,     6,    7,    8,    9,    10}}
+ *
+ * @throw  cudf::logic_error if source.num_columns() != target.num_columns()
+ * @throws cudf::logic_error if any `i`th source_column.type() != `i`th target_column.type()
+ * @throws cudf::logic_error if boolean_mask.type() != bool
+ * @throws cudf::logic_error if boolean_mask.size() != target.size()
+ *
+ * @param[in] source table_view (set of dense buffer) which gets scattered
+ * @param[in] target table_view which gets updated with scattered values from `source`
+ * @param[in] boolean_mask column_view which is a scatter_map.
+ * @param[in] mr Optional, The resource to use for all returned allocations
+ *
+ * @returns Returns a table by scattering `source` into `target` as per `boolean_mask`.
+ */
+std::unique_ptr<table> boolean_mask_scatter(
+    table_view const& source, table_view const& target,
+    column_view const& boolean_mask,
+    rmm::mr::device_memory_resource *mr = rmm::mr::get_default_resource());
+
+/**
```

Comment:
Do we need this version of the API? Could we use a single-column table input?

Reply:
What if the input table is of different type column, in that case scalar will be of different types.